hbase-server/src/main/java/org/apache/hadoop/hbase/HDFSBlocksDistribution.java

```diff
@@ -199,10 +225,23 @@ public long getUniqueBlocksTotalWeight() {
    * @return the locality index of the given host
    */
   public float getBlockLocalityIndex(String host) {
+    return getBlockLocalityIndex(host, false);
+  }
+  /**
+   * return the locality index of a given host
+   * @param host the host name
+   * @param forSsd only calc ssd type
+   * @return the locality index of the given host
+   */
+  public float getBlockLocalityIndex(String host, boolean forSsd) {
```

Comment:
As of now it seems fine, but do you think we might need more metrics similar to ssd? If so, this could be an enum input?
```
enum weightFor{  SSD, DEFAULT, ......(more params possible in future?) }
```
Thought?
Also, if there is no such chance for additional metrics, maybe we can name this method: `getBlockLocalityIndexWithSsd()` or something similar?

Reply:
Yeah, thought about it, but i think we wont have such chance for a long time, maybe rename is better.